force-app/main/default/classes/Chainable.cls

```diff
@@ -53,7 +80,49 @@ public abstract class Chainable {
         return sharedVariables.get(key);
     }
 
-	
+
+	public Chainable defer() {
+
+		List<ChainableDeferredEvent__e> deferEvents = unlink();
+
+		if(Limits.getLimitDMLStatements() < deferEvents.size()) {
+			handleDeferException(new DeferLimitException());
+		}
+
+		for(Database.SaveResult publishResult : EventBus.publish(deferEvents)) {
+			if(!publishResult.isSuccess()) {
+				handleDeferException(new DeferPublishException(publishResult.getErrors()));
+			}
+		}
+
+		return this;
+    }
+
+
+	public List<ChainableDeferredEvent__e> unlink() {
```

Comment:
I called it `unlink` since I though of it like it's "unlinking" the chain, meaning that every link in the chain will be published as a platform event, which is the process that makes deferring work, and therefore making the chain execute dynamically on the deferred context.

I could rename it to something more direct like `getDeferredChain` since it's called in the above `defer` method before publishing those